deps/oob-1.0.2/jsonutil.h

```diff
@@ -36,6 +36,50 @@ struct logDetails{
   int urgent;
 };
 
+struct dnsKey{
+    char description[16];
+    char driver[8];
+    char locale[8];
+    char host[8];
+    char port[8];
+    char account[8];
+    char user[8];
+    char password[8];
+    char token[8];
+    char authenticator[16];
+    char database[16];
+    char schema[8];
+    char warehouse[16];
+    char role[8];
+    char ssl[8];
+    char tracing[8];
+    char jwtTimeout[16];
+    char timezone[16];
+    char privKey[16];
+};
+
+struct dnsVal{
+    char description[4096];
+    char driver[64];
+    char locale[64];
+    char host[512];
+    char port[10];
+    char account[256];
+    char user[256];
+    char password[8];
+    char token[1024];
+    char authenticator[1024];
+    char database[256];
+    char schema[256];
+    char warehouse[256];
+    char role[256];
+    char ssl[10];
+    char tracing[10];
+    char jwtTimeout[256];
+    char timezone[64];
+    char privKey[1024];
+};
+
```

Comment:
```suggestion
enum {KEY=0,VALUE=1};

#define MAX_KEY 40
#define MAX_VAL 1024

typedef struct keyVal{
  char key[MAX_KEY];
  char val[MAX_VAL];
}keyval;

typedef struct dsnfields
{
  //You can define in a single array but then again you need another enum
  keyval description;
  keyval driver;
  keyval host;
  keyval port;
  //Add more
} dsnFields;
//dsnFields dsn ; 
//You can access them like dsn.driver.key 
```

Reply:
I was thinking about this too but I was just afraid of setting a general memory limit for each field. For example description or authenticator need 1024 but most cases require far less so I thought it'd be less efficient memory wise. Also the existing code also seems to point to customizing the memory limit but this can be changed. @sfc-gh-sshankar  What do you think?

Reply:
Your event message is short lived. 
variable should be a local and goes out of scope after the oob event is sent, thus freeing the stack memory.
And for DSN this is done once at the time of connection.

Reply:
Can you also check the oob event with dsn fields is json parsable. 
one way is to check some online parsers.
When i checked the event in the description it was not parsable.

Reply:
I've said this privately but if you're just copy-pasting the GitHub description there are errors because the single quote (') is recognized differently.